flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterSchedulerTest.java

```diff
@@ -72,16 +70,13 @@ public void testIfStartSchedulingFailsJobMasterFailsFatally() throws Exception {
 		System.setSecurityManager(trackingSecurityManager);
 
 		final SchedulerNGFactory schedulerFactory = new FailingSchedulerFactory();
-		final JobMaster jobMaster = new JobMasterBuilder(new JobGraph(), TESTING_RPC_SERVICE_RESOURCE
-			.getTestingRpcService())
+		final JobMaster jobMaster = new JobMasterBuilder(new JobGraph(), TESTING_RPC_SERVICE_RESOURCE.getTestingRpcService())
 			.withSchedulerFactory(schedulerFactory)
 			.createJobMaster();
 
-		final CompletableFuture<Acknowledge> startFuture = jobMaster.start(JobMasterId.generate());
+		jobMaster.start();
 
 		try {
-			startFuture.join();
-
 			assertThat(trackingSecurityManager.getSystemExitFuture().join(), is(FatalExitExceptionHandler.EXIT_CODE));
```

Comment:
This test got stuck on CI; I suppose we never call the fatal error handler since an error during the start of the scheduling now prevents the RpcEndpoint from starting altogether.

Reply:
The signal is now sent via the `OnCompletionActions` and no longer via the `SecurityManager`. I will update this test case.